packages/devtools_app/lib/src/app.dart

```diff
@@ -141,12 +141,23 @@ class DevToolsAppState extends State<DevToolsApp> with AutoDisposeMixin {
       }
     });
     addAutoDisposeListener(serviceConnection.serviceManager.connectedState, () {
-      final connected =
-          serviceConnection.serviceManager.connectedState.value.connected;
+      final connectionState =
+          serviceConnection.serviceManager.connectedState.value;
+
+      // When disconnecting from an app, prepare the offline state for reviewing
+      // history.
+      screenControllers.forEachInitialized((screenController) {
+        if (screenController is OfflineScreenControllerMixin &&
```

Comment:
small suggestion: Add a getter to `ScreenController` (e.g. `hasOfflineSupport`) so that we we don't have to do `OfflineScreenControllerMixin` type check